sql/core/src/main/scala/org/apache/spark/sql/execution/pythonUDFs.scala

```diff
@@ -338,7 +338,11 @@ case class BatchPythonEvaluation(udf: PythonUDF, output: Seq[Attribute], child:
   def children: Seq[SparkPlan] = child :: Nil
 
   protected override def doExecute(): RDD[InternalRow] = {
-    val childResults = child.execute().map(_.copy())
+    val buffer = new java.util.concurrent.ConcurrentLinkedQueue[InternalRow]()
```

Comment:
If i understand this correctly, we are assuming the following in order for this to work:

1. Each task gets their own copy of the deserialized closure, and thus their own copy of the queue.
2. All closures are serialized together in one shot, rather than in multiple places (e.g. they are all done in the serializer, not in the ctor of the RDD)
3. Java serializer does not serialize objects twice within the same stream, since it uses it to detect cycles. When they are deserialized, they still point to the same copy.